include/zephyr/mgmt/mcumgr/callbacks.h

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef H_MCUMGR_CALLBACKS_
+#define H_MCUMGR_CALLBACKS_
+
+#include <inttypes.h>
+#include <zephyr/sys/slist.h>
+#include <mgmt/mgmt.h>
+
+#ifdef CONFIG_MCUMGR_CMD_FS_MGMT
+#include <zephyr/mgmt/mcumgr/fs_mgmt_callbacks.h>
+#endif
+
+#ifdef CONFIG_MCUMGR_CMD_IMG_MGMT
+#include <zephyr/mgmt/mcumgr/img_mgmt_callbacks.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief MCUmgr callback API
+ * @defgroup mcumgr_callback_api MCUmgr callback API
+ * @ingroup mcumgr
+ * @{
+ */
+
+/**
+ * @typedef mgmt_cb
+ * @brief Function to be called on MGMT notification/event.
+ *
+ * This callback function is used to notify an application or system about a mcumgr mgmt event.
+ *
+ * @param event		MGMT_EVT_OP_[...].
+ * @param failed	Will be set to true if a previous handler has already returned a failure
+ *			error code (i.e. this handler is being called for a notification only,
+ *			the return code will be ignored)
+ * @param abort_more	Set to true to abort further processing by additional handlers.
+ * @param data		Optional event argument.
+ * @param data_size	Size of optional event argument (0 if no data is provided).
+ *
+ * @return		MGMT_ERR_[...] of the status to return to the calling code (only checked
+ *			when failed is false).
+ */
+typedef int (*mgmt_cb)(uint32_t event, bool failed, bool *abort_more, void *data,
+		       size_t data_size);
+
+/** Event ID which signfies that all events for a particular group should be enabled. */
+#define MGMT_EVT_OP_ID_ALL 0xffff
+
+/** Get combined ID for a particular group and event ID. */
+#define MGMT_EVT_OP_ID(group, id) ((group << 16) | BIT(id))
+
+/** Get combined ID used for enabling all events of a particular group. */
+#define MGMT_EVT_OP_ALL(group) ((group << 16) | MGMT_EVT_OP_ID_ALL)
+
+/** Get group from combined ID. */
+#define MGMT_EVT_GET_GROUP(event) ((event >> 16) & MGMT_EVT_OP_ID_ALL)
+
+/** Get event ID from combined ID. */
+#define MGMT_EVT_GET_ID(event) (event & MGMT_EVT_OP_ID_ALL)
+
+/**
+ * MGMT event callback group IDs. Note that this is not a 1:1 mapping with MGMT_GROUP_ID_[...]
+ * values.
+ */
+enum mgmt_callback_groups {
+	MGMT_CB_GROUP_ALL = 0,
```

Comment:
`MGMT_EVT_CB` or `MGMT_EVNT_GRP`, you have basically created the `MGMT_EVT` namespace, so lets use it.